hpx/plugins/parcel/message_buffer.hpp

```diff
@@ -60,7 +60,27 @@ namespace hpx { namespace plugins { namespace parcel { namespace detail
         void operator()(parcelset::parcelport* pp)
         {
             if (!messages_.empty())
+            {
+                if (0 == threads::get_self_ptr())
```

Comment:
Hmmm, going through the parcel handler again would force the created (coalesced) parcels to go through coalescing again - not sure if we can prevent this. 

Also, in case of migration, the destination will resolve this just fine. 

In some sense, the parcel-coalescing _is_ a parcel-handler on its own...